pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -402,6 +375,51 @@ pub mod pallet {
 				.collect::<Vec<_>>();
 			Dna::try_from(dna).map_err(|_| Error::<T>::IncorrectDna.into())
 		}
+
+		pub(crate) fn do_mint(player: &T::AccountId, how_many: MintCount) -> DispatchResult {
+			ensure!(Self::mint_available(), Error::<T>::MintUnavailable);
+
+			let current_block = <frame_system::Pallet<T>>::block_number();
+			if let Some(last_block) = Self::last_minted_block_numbers(&player) {
+				let cooldown = Self::mint_cooldown();
+				ensure!(current_block > last_block + cooldown, Error::<T>::MintCooldown);
+			}
+
+			let how_many = how_many as usize;
+			let max_ownership = (MAX_AVATARS_PER_PLAYER as usize)
+				.checked_sub(how_many)
+				.ok_or(ArithmeticError::Underflow)?;
+			ensure!(Self::owners(&player).len() <= max_ownership, Error::<T>::MaxOwnershipReached);
```

Comment:
This checks whether there's enough capacity for `player` to mint `how_many` number of avatars.